### ISSUE TITLE
Passing commandline arguments to an SBT script doesn't work in 0.13.1

### ISSUE THREAD

**Issue by 404-** [sbt/sbt#1041]
Given an executable file called 'sbt-test.scala' which contains the code below, attempting to run using SBT 0.13.1 with the following args yields errors:

```
$ ./sbt-test.scala --arg=val
[error] Expected ID character
[error] Not a valid command: arg
[error] arg=val
```

```
$ ./sbt-test.scala  --flag --arg=val
[error] Not a valid command: flag
[error] flag
```

Running the same commands but with SBT 0.13.0 completes successfully and yields expected output without any errors

``` scala
#!/bin/sh
SBT_0_13_0="/opt/tripadvisor/l/data/multilib/sbt/0.13.0/sbt-launch-0.13.0.jar"
SBT_0_13_1="/opt/tripadvisor/l/data/multilib/sbt/0.13.1/sbt-launch-0.13.1.jar"
SBT="$SBT_0_13_1"
exec java -Dsbt.main.class=sbt.ScriptMain -jar "$SBT" $0 "$@"
!#

/***
scalaVersion := "2.10.3"

libraryDependencies ++= Seq(
  "com.typesafe" % "config" % "1.0.0"
)
*/

import com.typesafe.config.{ConfigFactory, Config}

println(s"Args: ${args mkString ", "}")

val cfg = ConfigFactory.parseString(
  """
    |credentials {
    |  user = someone
    |  pass = s3cr3t
    |}
  """.stripMargin)

println(cfg getString "credentials.user")
```

**Comment by dwijnand** [comment]
This must've broke with the 0.13.1 feature:

> Early commands: any command prefixed with -- gets moved before other commands on startup and doesn’t force sbt into batch mode.

[0.13.1 release notes](http://www.scala-sbt.org/0.13/docs/ChangeSummary_0.13.0.html#sbt+0.13.1)

**Comment by nafg** [comment]
+1

**Comment by bshlgrs** [comment]
Any updates to this?

**Comment by nafg** [comment]
Have you tried on 0.13.9?

**Comment by o-martin** [comment]
Pb is still there with 0.13.9 all arguments starting with "--" are interpreted by sbt and not passed to the scala scripts as arguments. Other pattern are successfully passed

**Comment by SethTisue** [comment]
Seems like this would be a fix that someone out there in open-source land could reasonably tackle without having to learn tons about how sbt works? I assume that's why @jsuereth tagged it "community".

Maybe a bare `--` option should terminate the special handling and signal that remaining flags would be passed to the script? I haven't thought it through, but I've seen this in other command-line tools (e.g. `grep`).

**Comment by eed3si9n** [comment]
+1 on what @SethTisue said.

**Comment by hntd187** [comment]
Write a rather large script with some deps in it and experiencing same thing with sbt 0.13.9, switching to 0.13.0 and the script works perfectly fine.

+1

**Comment by eed3si9n** [comment]
This will be fixed in sbt 1.0 via #2741

**Comment by dgoetsch** [comment]
This is why I am never using sbt again.
